### PR TITLE
New features!

### PR DESCRIPTION
**Pull request by eshleebien**
- Added manage my URL
- Added sys admin
- Added filter on URL display

### CODE REVIEW

File: app/Http/Controllers/Api/UserController.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+use Carbon\Carbon;
+use App\Http\Requests;
+use App\Http\Controllers\Controller;
+use App\Http\Requests\UpdateRedirectRequest;
+
+class UserController extends Controller
+{
+    public function __contruct()
+    {
+        $this->middleware('auth');
```
Comment (Jhoem) [inline app/Http/Controllers/Api/UserController.php]: put anything like this on the routes and use route groups :)

File: app/Http/routes.php [hunk 2]
```diff
@@ -13,7 +13,8 @@
 
 Route::get('/', 'RedirectController@index');
 Route::controller('/au', 'Auth\AuthController');
```
Comment (Jhoem) [inline app/Http/routes.php:15]: use route group

e.g

```
Route::group(['middleware' => 'auth'], function () {
    Route::get('/', function ()    {
        // Uses Auth Middleware
    });

    Route::get('user/profile', function () {
        // Uses Auth Middleware
    });
});
```

File: app/Http/Controllers/UserController.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+use Session;
+use App\Http\Requests;
+use App\Http\Controllers\Controller;
+
+class UserController extends Controller
+{
+    public function __construct()
+    {
+        $this->middleware('auth');
```
Comment (Jhoem) [inline app/Http/Controllers/UserController.php]: pin

File: app/Http/Controllers/Api/UserController.php [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            }
+            else {
+                $links->where('email', 'like', '%' . $queryString . '%')
+                    ->orWhere('short_url', 'like', '%' . $queryString . '%')
+                    ->orWhere('long_url', 'like', $queryString . '%');
+            }
+
+        }
+
+        $links = $links->paginate(50);
```
Comment (wildtron) [inline app/Http/Controllers/Api/UserController.php:47]: please use vanilla implementation of pagination

File: app/Http/Middleware/VerifyCsrfToken.php [hunk 5]
```diff
@@ -3,6 +3,7 @@
 namespace App\Http\Middleware;
 
 use Closure;
+use App;
```
Comment (wildtron) [inline app/Http/Middleware/VerifyCsrfToken.php:5]: please add new line after this

File: app/Http/Controllers/UserController.php [hunk 6]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+
+use Session;
```
Comment (wildtron) [inline app/Http/Controllers/UserController.php:5]: single words first

File: app/Http/Requests/UpdateRedirectRequest.php [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'long_url' => 'required:url',
+            'short_url' => 'required|min:2',
```
Comment (wildtron) [inline app/Http/Requests/UpdateRedirectRequest.php]: let's use array la `=> [ 'required', 'min:2']`

File: resources/assets/js/app.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+        $scope.deleteLink = function (link) {
+            link._token = _token;
+
+            $http({
+                url: '/api/my/links',
+                method: 'DELETE',
+                params: link
+            }).success(function (data) {
+                $scope.links.data.splice(link.position, 1);
+            });
```
Comment (wildtron) [inline resources/assets/js/app.js:74]: please add `.error(function () {});`

**Pull request merged**